Flippers/src/main/java/io/doist/recyclerviewext/flippers/EmptyRecyclerFlipper.java

```diff
@@ -45,36 +67,129 @@ public boolean monitor(RecyclerView.Adapter adapter) {
                 mAdapter.registerAdapterDataObserver(mObserver);
                 mCount = mAdapter.getItemCount();
                 if (mCount > 0) {
-                    replaceNoAnimation(mEmptyView, mRecyclerView);
+                    showContent(false);
                 } else {
-                    replaceNoAnimation(mRecyclerView, mEmptyView);
+                    showEmpty(false);
                 }
             }
+
             return true;
         }
+
         return false;
     }
 
-    private void disableItemAnimatorForRecyclerInAnimation() {
-        final RecyclerView.ItemAnimator itemAnimator = mRecyclerView.getItemAnimator();
-        FlipperAnimator flipperAnimator = getFlipperAnimator();
-        if (itemAnimator != null && flipperAnimator != null) {
-            mRecyclerView.setItemAnimator(null);
-            mRecyclerView.postDelayed(new Runnable() {
+    protected void showContent(boolean animate) {
+        executeFlip(mEmptyView, mRecyclerView, animate);
+    }
+
+    protected void showEmpty(boolean animate) {
+        executeFlip(mRecyclerView, mEmptyView, animate);
+    }
+
+    protected void onFlipCompleted(View outView, View inView, boolean animate) {
+        if (animate && inView == mRecyclerView) {
+            disableItemAnimatorForRecyclerInAnimation();
+        }
+    }
+
+    protected void executeFlip(final View outView, final View inView, final boolean animate) {
+        if (animate) {
+            mFlipper.replace(outView, inView, new Runnable() {
                 @Override
                 public void run() {
-                    mRecyclerView.setItemAnimator(itemAnimator);
+                    onFlipCompleted(outView, inView, true);
+                }
+            });
+        } else {
+            mFlipper.replaceNoAnimation(outView, inView, new Runnable() {
+                @Override
+                public void run() {
+                    onFlipCompleted(outView, inView, false);
                 }
-            }, flipperAnimator.getDuration());
+            });
         }
     }
 
+    protected RecyclerView getRecyclerView() {
+        return mRecyclerView;
+    }
+
+    protected View getEmptyView() {
+        return mEmptyView;
+    }
+
+    protected Flipper getFlipper() {
+        return mFlipper;
+    }
+
+    /**
+     * @deprecated To customize the flipper use the {@link #getFlipper()}.
+     */
+    @Override
+    public FlipperAnimator getFlipperAnimator() {
+        return mFlipper.getFlipperAnimator();
+    }
+
+    /**
+     * @deprecated To customize the flipper use the {@link #getFlipper()}.
+     */
+    @Override
+    public void setFlipperAnimator(FlipperAnimator flipperAnimator) {
+        mFlipper.setFlipperAnimator(flipperAnimator);
+    }
+
+    /**
+     * @deprecated To customize the flipper use the {@link #getFlipper()}.
```

Comment:
Why we deprecate methods `replace*`? They are not about cusomization :)

Reply:
I was deprecating these methods so we could remove the subclassing from `Flipper` but as you said:

> I think choice subclassing vs composing was made when this library was implemented. I think it's late to change this 😅

I've just let them as an interface for the inner `Flipper` without warning.